app/src/main/java/ch/epfl/sweng/SDP/matchmaking/Matchmaker.java

```diff
@@ -1,73 +1,131 @@
             }
 
             @Override
-            public void onCancelled(@NonNull DatabaseError error) {
+            public void onCancelled(DatabaseError error) {
                 // Failed to read value
             }
         });
+
+
     }
 
     /**
-     * Join a room.
-     *
-     * @param roomId the id of the room.
+     * join a room.
      */
-    public void joinRoom(String roomId) {
-        // FirebaseUser currentFirebaseUser = FirebaseAuth.getInstance().getCurrentUser() ;
-        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(myRef);
-        builder.addChildren(roomId + ".users.123").build().setValue("InRoom");
+    public Boolean joinRoom() {
+
+            Boolean successful = false;
+        HttpURLConnection connection = null;
+
+        try {
+                //Create connection
+                String urlParameters = "userId=" + URLEncoder.encode(constantsWrapper.getFirebaseUserId(), "UTF-8");
+                URL url = new URL("https://us-central1-gyrodraw.cloudfunctions.net/joinGame?" + urlParameters);
+                connection = createConnection(url);
+
+                //Send request
+                DataOutputStream wr = new DataOutputStream(
+                        connection.getOutputStream());
+                wr.close();
+
+                if (connection.getResponseCode() == HttpURLConnection.HTTP_OK) {
+                    // OK
+                    successful = true;
+                    // otherwise, if any other status code is returned, or no status
+                    // code is returned, do stuff in the else block
+                }
+            } catch (Exception e) {
+                successful = false;
+                e.printStackTrace();
+            } finally {
+                if (connection != null) {
+                    connection.disconnect();
+                }
+            }
+        return successful;
+    }
+
+    /**
+     * Creates a connection.
+     * @return set up connection
+     */
+    private HttpURLConnection createConnection(URL url) throws IOException {
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Content-Type",
+                "application/x-www-form-urlencoded");
+
+        connection.setRequestProperty("Content-Language", "en-US");
+
+        connection.setUseCaches(false);
+        connection.setDoOutput(true);
+        return connection;
     }
 
     /**
-     * Leave a room.
-     *
+     * leave a room.
      * @param roomId the id of the room.
      */
-    public void leaveRoom(String roomId) {
-        // FirebaseUser currentFirebaseUser = FirebaseAuth.getInstance().getCurrentUser() ;
-        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(myRef);
-        builder.addChildren(roomId + ".users.123").build().removeValue();
+    public Boolean leaveRoom(String roomId) {
```

Comment:
The leaveRoom function has a couple of problems. First, it doesn't handle the case where the deletion fails (it always returns true), and second, Firebase functions run asynchronously. This means that you should only continue once Firebase answered (i.e. put code to execute next into a listener). That's something you'll have to account for in the future.

Reply:
Yes you are right, I should use a completion handler here.